### ISSUE TITLE
Strong DB dependency

### ISSUE THREAD

**Issue by dhiana** [jphalip/django-treemenus#32]
The show_menu template tag performs Menu.objects.get(...). If the database is not populated (for example, when running tests), any view referencing a template with this templatetag (which is usually at base template, affecting all pages!) will raise DoesNotExist.

Shouldn't the menu fail gracefully by not showing anything (just like django's own templatetags) instead of allowing this exception?

**Comment by jphalip** [comment]
Yes, I agree it should fail more gracefully and only show the exception if `TEMPLATE_DEBUG` (https://docs.djangoproject.com/en/dev/ref/settings/#template-debug) is set to True.

Would you like to provide a patch with some tests?

**Comment by dhiana** [comment]
Sure!

**Comment by dhiana** [comment]
```
$ make test
DJANGO_SETTINGS_MODULE=treemenus.tests.settings django-admin.py test treemenus
Traceback (most recent call last):
  File "/home/dhiana/.virtualenvs/dj16/bin/django-admin.py", line 5, in <module>
    management.execute_from_command_line()
  File "/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/core/management/__init__.py", line 399, in execute_from_command_line
    utility.execute()
  File "/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/core/management/__init__.py", line 392, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/core/management/__init__.py", line 261, in fetch_command
    commands = get_commands()
  File "/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/core/management/__init__.py", line 107, in get_commands
    apps = settings.INSTALLED_APPS
  File "/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/conf/__init__.py", line 54, in __getattr__
    self._setup(name)
  File "/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/conf/__init__.py", line 49, in _setup
    self._wrapped = Settings(settings_module)
  File "/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/conf/__init__.py", line 132, in __init__
    % (self.SETTINGS_MODULE, e)
ImportError: Could not import settings 'treemenus.tests.settings' (Is it on sys.path? Is there an import error in the settings file?): No module named treemenus.tests.settings
make: *** [test] Error 1
```

I've tried django 1.4 and 1.6. Any hints?

**Comment by jphalip** [comment]
Where did you run the 'make test' command from? Have you tried from the repository's root?

**Comment by dhiana** [comment]
I'm on the repo root. Calling python and importing the settings module
works fine...

```
(dj16)[dhiana@csrjl06l django-treemenus]$ ls
AUTHORS.txt  docs  LICENCE.txt  Makefile  README.rst  setup.cfg  setup.py
 treemenus

(dj16)[dhiana@csrjl06l django-treemenus]$ make test
DJANGO_SETTINGS_MODULE=treemenus.tests.settings django-admin.py test
treemenus
Traceback (most recent call last):
  File "/home/dhiana/.virtualenvs/dj16/bin/django-admin.py", line 5, in
<module>
    management.execute_from_command_line()
  File
"/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/core/management/__init__.py",
line 399, in execute_from_command_line
    utility.execute()
  File
"/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/core/management/__init__.py",
line 392, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File
"/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/core/management/__init__.py",
line 261, in fetch_command
    commands = get_commands()
  File
"/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/core/management/__init__.py",
line 107, in get_commands
    apps = settings.INSTALLED_APPS
  File
"/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/conf/__init__.py",
line 54, in __getattr__
    self._setup(name)
  File
"/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/conf/__init__.py",
line 49, in _setup
    self._wrapped = Settings(settings_module)
  File
"/home/dhiana/.virtualenvs/dj16/lib/python2.7/site-packages/django/conf/__init__.py",
line 132, in __init__
    % (self.SETTINGS_MODULE, e)
ImportError: Could not import settings 'treemenus.tests.settings' (Is it on
sys.path? Is there an import error in the settings file?): No module named
treemenus.tests.settings
make: *** [test] Error 1

(dj16)[dhiana@csrjl06l django-treemenus]$ python
Python 2.7.5 (default, Jun 26 2014, 11:55:39)
[GCC 4.8.2 20131212 (Red Hat 4.8.2-7)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import treemenus.tests.settings
```

2014-08-18 17:51 GMT-03:00 Julien Phalip notifications@github.com:

**Comment by jphalip** [comment]
Could you try this?

```
$ PYTHONPATH=. make test
```

**Comment by dhiana** [comment]
It worked! Any thoughts on why this is necessary?

2014-09-06 16:05 GMT-03:00 Julien Phalip notifications@github.com:

**Comment by jphalip** [comment]
Fixed in 4b8882fea180079838c18af58d68dfe9547881ed. Thank you!
